libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -215,6 +216,42 @@ std::string YulUtilFunctions::copyLiteralToStorageFunction(std::string const& _l
 	});
 }
 
+std::string YulUtilFunctions::revertWithError(
+	std::string const& _signature,
+	std::vector<Type const*> const& _parameterTypes,
+	std::vector<ASTPointer<Expression const>> const& _errorArguments,
+	std::string const& _pos,
+	std::string const& _end
+)
+{
+	solAssert((!_pos.empty() && !_end.empty()) || (_pos.empty() && _end.empty()));
+	bool const needsNewVariable = !_pos.empty() && !_end.empty();
+
+	Whiskers templ(R"({
+		let <pos> := <allocateUnbounded>()
+		mstore(<pos>, <hash>)
+		let <end> := <encode>(add(<pos>, 4) <argumentVars>)
+		revert(<pos>, sub(<end>, <pos>))
+	})");
```

Comment:
I think that these braces would make more sense in `requireWithErrorFunction()`, as a part of the `if`. This way you'd keep the original uses of this function unchanged.

```suggestion
	Whiskers templ(R"(
		let <pos> := <allocateUnbounded>()
		mstore(<pos>, <hash>)
		let <end> := <encode>(add(<pos>, 4) <argumentVars>)
		revert(<pos>, sub(<end>, <pos>))
	)");
```

Reply:
How would that keep the original uses of the function unchanged, if I change the original function? `revertWithError` is used in other places in `IRGeneratorForStatements`, which would basically remove the scoping block from those if I were to apply your change.

Reply:
Ah, wait. I somehow wrongly remembered from previous reviews that the original code was *not* wrapped in a block. And it made sense to be because otherwise if it were, why would it need globally unique names for the variables?

But looking at it now, the braces were actually there, so you're right.

In that case I'd just suggest making the brace a bit more prominent so that it's easy to see it's there:

```suggestion
	Whiskers templ(R"(
		{
			let <pos> := <allocateUnbounded>()
			mstore(<pos>, <hash>)
			let <end> := <encode>(add(<pos>, 4) <argumentVars>)
			revert(<pos>, sub(<end>, <pos>))
		}
	)");
```

Reply:
I did that, but then the generated IR looks wonky, i.e.

```
function revertWithError(args)
       
             { 
                    let <pos> := <allocateUnbounded>()
                    // ....
              }
```

I can do it if it doesn't matter for e.g. auditing?

Reply:
Ah, ok, that's not ideal. I assumed it would not change actually generated code, but you're right, of course we don't trim it in any way so it will look wonky.

Let's leave it as is then. If the simplest improvement didn't work as expected, it's not worth putting more thought into it.